internal/transports/p2p/server.go

```diff
@@ -56,9 +72,31 @@ func (s *server) Start() error {
 		return err
 	}
 
-	s.peer = peer
+	return s.connectPeer(peer)
+}
+
+func (s *server) listenAndConnect() error {
+	s.log.Info().Msgf("listening for inbound connections on port %s", s.chainParams.DefaultPort)
 
-	err = peer.Connect()
+	ourAddr := net.JoinHostPort("", s.chainParams.DefaultPort)
+	listener, err := net.Listen("tcp", ourAddr)
+	if err != nil {
+		s.log.Error().Msgf("error creating listener, reason: %v", err)
+		return err
+	}
+
+	peer, err := peer.NewInboundPeer(listener, s.config, s.chainParams, s.headersService, s.chainService, s.log)
+	if err != nil {
+		return err
+	}
+
+	return s.connectPeer(peer)
+}
+
+func (s *server) connectPeer(peer *peer.Peer) error {
+	s.peers = append(s.peers, peer)
```

Comment:
Is it designed to be executed from single goroutine or concurrently from many?

Reply:
For now it's synchronous, but it should be rewritten to support concurrency in the next task regarding multiple connections.

But if you think I should support concurrency here already in this PR, let me know :)

Reply:
IMHO this is good moment to at least handle unlimited number of incoming connections. And just put adding some limits and managing list of those connections to the other task.
Because there are not so many code changes so I can predict that additional for loop couldn't make it hard to review, but I assume that adding this in a later stage, could make the later PR harder to read and actually it can be easily skipped because outbound connections have higher priority for us.